python/src/scipp/neutron/diffraction/load.py

```diff
@@ -56,39 +56,28 @@ def load_calibration(filename, mantid_LoadDiffCal_args={}):
     cal_data["difa"].unit = sc.units.us / sc.units.angstrom / sc.units.angstrom
     cal_data["tzero"].unit = sc.units.us
 
-    # Mask data not used, but is loaded by LoadDiffCal
-    # output.OutputMaskWorkspace)
-
-    # Apply group information to dataset by matching detector id's to group nr
+    mask_ws = output.OutputMaskWorkspace
     group_ws = output.OutputGroupingWorkspace
-    group_map = {}  # dict from detectorID to group number
-    spectrum_info = group_ws.spectrumInfo()
-    detector_info = group_ws.detectorInfo()
-    det_ids = detector_info.detectorIDs()
-    for i, spec in enumerate(spectrum_info):
-        spec_def = spec.spectrumDefinition
-        # We take the first detector in the definition, because that's
-        # what Mantid does via DetectorGroup::getID. Id is first det of group.
-        definition = spec_def[0]
-        # Discard time index
-        det_index = definition[0]
-        group_map[det_ids[det_index]] = group_ws.readY(i)[0]
-
-    # Create list with same ordering as in the cal_data dataset
-    cal_det_ids = cal_data["detid"].values
-    group_list = np.fromiter((group_map[detid] for detid in cal_det_ids),
-                             count=len(cal_det_ids),
-                             dtype=np.int32)
-
-    cal_data["group"] = sc.Variable([sc.Dim.Row], values=group_list)
+    rows = mask_ws.getNumberHistograms()
+    mask = np.fromiter((mask_ws.readY(i)[0] for i in range(rows)),
+                       count=rows,
+                       dtype=np.bool)
+    group = np.fromiter((group_ws.readY(i)[0] for i in range(rows)),
```

Comment:
Maybe add a comment that mentions what you say in the PR description.
Something like:
"There is no need here to handle potentially mismatched ordering, since inputs are from an algorithm that should guarantee this."

Reply:
See update.